lib/octicons_react/src/index.js

```diff
@@ -6,7 +6,7 @@ export default function Octicon({icon: Icon, children, ...props}) {
     // eslint-disable-next-line github/unescaped-html-literal
     '<Octicon> is deprecated. Use icon components on their own instead (e.g. <Octicon icon={AlertIcon} /> → <AlertIcon />)'
   )
-  return typeof Icon === 'function' ? <Icon {...props} /> : React.cloneElement(React.Children.only(children), props)
+  return typeof Icon === 'object' ? <Icon {...props} /> : React.cloneElement(React.Children.only(children), props)
```

Comment:
I am a little unsure about this change 😵‍💫 Would there be any impact on consumer applications? 🤔

Reply:
Thanks for your great suggestion @joshblack! I removed the deprecated `Octicons` 🎉  Build and tests seems fine but let me know if there is anything I am missing to do here.

Also, I pushed it as a commit here, would that be better if I pushed another PR for it you think? 🤔

Reply:
If it is good to go within this PR, I'll update the title and the changeset. If not, I'll push another PR